spec/ruby/core/module/define_method_spec.rb

```diff
@@ -631,3 +631,28 @@ def m(a, b, *c)
     @obj.method(:n).parameters.should == @obj.method(:m).parameters
   end
 end
+
+describe "Method#define_method when passed a Proc object" do
+  describe "and a method is defined inside" do
+    it "defines the nested method in the default definee where the Proc was created" do
+      prc = nil
+      t = Class.new do
+        prc = -> {
+          def nested_method_in_proc_for_define_method
+            42
+          end
+        }
+      end
```

Comment:
Some of these edge cases are insane.

Reply:
Indeed. And the default definee in Ruby can only be observed through where the method gets defined, which doesn't help in making tests/specs clear.